From 328c711017659c76283a81661789d27ec19e517b Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Wed, 5 Sep 2018 14:50:10 +0800
Subject: [PATCH 216/741] fmd: use "fsl,fman-ptp-timer" compatible for ptp
 probe

commit 9a56c6f883cd5d6b95f33cb4436c79dba2b7e229 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Current ptp compatible "fsl,fman-rtc" used for ptp probe
in fmd driver couldn't involve PowerPC DPAA FMan PTP timer.
Let's use "fsl,fman-ptp-timer" instead to support DPAA FMan
PTP timer of both ARM and PowerPC.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 .../net/ethernet/freescale/sdk_fman/src/wrapper/lnxwrp_fm.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/freescale/sdk_fman/src/wrapper/lnxwrp_fm.c b/drivers/net/ethernet/freescale/sdk_fman/src/wrapper/lnxwrp_fm.c
index 31f654b4c617..c5fc266e534c 100755
--- a/drivers/net/ethernet/freescale/sdk_fman/src/wrapper/lnxwrp_fm.c
+++ b/drivers/net/ethernet/freescale/sdk_fman/src/wrapper/lnxwrp_fm.c
@@ -687,9 +687,9 @@ static t_LnxWrpFmDev * ReadFmDevTreeNode (struct platform_device *of_dev)
     if (WARN_ON(strlen("ptp-timer") >= sizeof(ids[0].name)))
         return NULL;
     strcpy(ids[0].name, "ptp-timer");
-    if (WARN_ON(strlen("fsl,fman-rtc") >= sizeof(ids[0].compatible)))
+    if (WARN_ON(strlen("fsl,fman-ptp-timer") >= sizeof(ids[0].compatible)))
         return NULL;
-    strcpy(ids[0].compatible, "fsl,fman-rtc");
+    strcpy(ids[0].compatible, "fsl,fman-ptp-timer");
     for_each_child_of_node(fm_node, dev_node) {
         if (likely(of_match_node(ids, dev_node) != NULL)) {
             _errno = of_address_to_resource(dev_node, 0, &res);
@@ -920,7 +920,7 @@ static t_Error ConfigureFmDev(t_LnxWrpFmDev  *p_LnxWrpFmDev)
 
     if (p_LnxWrpFmDev->fmRtcPhysBaseAddr)
     {
-        dev_res = __devm_request_region(p_LnxWrpFmDev->dev, p_LnxWrpFmDev->res, p_LnxWrpFmDev->fmRtcPhysBaseAddr, p_LnxWrpFmDev->fmRtcMemSize, "fman-rtc");
+        dev_res = __devm_request_region(p_LnxWrpFmDev->dev, p_LnxWrpFmDev->res, p_LnxWrpFmDev->fmRtcPhysBaseAddr, p_LnxWrpFmDev->fmRtcMemSize, "fman-ptp-timer");
         if (unlikely(dev_res == NULL))
             RETURN_ERROR(MAJOR, E_INVALID_STATE, ("__devm_request_region() failed"));
 
-- 
2.17.1

